index.js

```diff
@@ -57,6 +73,34 @@ router
     ctx.status = 202
     ctx.body = 'OK'
   })
+  .get('/cloudflareError.png', async ctx => {
+    if (ctx.query['cloudflareErrorType'] == undefined || ctx.query['rayID'] == undefined || ctx.query['clientIP'] == undefined) {
+      ctx.status = 404
+    }
+    else if (!['500', '1000'].includes(ctx.query['cloudflareErrorType'])) {
+      console.error('Unregistered error type')
+      ctx.status = 404
+    }
+    else if (!rayIDRegex.test(ctx.query['rayID'])) {
+      console.error('Invalid Ray ID')
+      ctx.status = 404
+    }
+    else if (!new ipAddress(ctx.query['clientIP']).isValid()) {
+      console.error('Invalid client IP')
+      ctx.status = 404
+    }
+    else {
+      postMetric("volley", {
+        "type": "increment",
+        "name": "cloudflareError",
+        "sampleRate": 1,
+        "tags": ["cloudflareErrorType:" + ctx.query['cloudflareErrorType']]
+      })
```

Comment:
Is this a synchronous or asynchronous task?

Reply:
Will leave it as synchronous for now as per our conversation

Reply:
(Which, for posterity, was that making these async would make them slightly harder to test, not that it might not actually be worth it to move these out of the request cycle.)